src/archive/src/main.rs

```diff
@@ -239,6 +259,84 @@ fn write_entry(anchor: AnchorNumber, timestamp: Timestamp, entry: ByteBuf) {
             ))
         }
     });
+    write_entry_internal(anchor_number, timestamp, entry)
+}
+
+/// Fetches, archives and acknowledges a batch of entries.
+/// *Note:* Must be written in a way that nothing breaks on overlapping executions of [fetch_entries].
```

Comment:
what do you mean by this?

> in a way that nothing breaks on overlapping executions of [fetch_entries]

Can you clarify why we'd have overlapping executions? Also, why not just protect this with a mutex or lock?

I'm guessing this stems from user the Timer API? Can you shed some light on how this work, also w.r.t. the "last successful call" metric?

Reply:
Ok, if checked with the message routing team and apparently it is possible for responses to get reordered. This means that we could potentially benefit from having overlapping calls because the call started later could finish before the other call.

The "last successful call" metric would always show the timestamp of the last completed invocation of the poll. From this we could infer a healthy system if the timestamp changes frequently enough and is reasonably recent.